### PR TITLE
Add e2e test vscode workspace with more flexible launch targets

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Adds a VSCode workspace set up for e2e test debugging. It has the following advantages over the primary (root) workspace:

- Access to legacy module installation from the file explorer
- Dedicated launch configurations for each driver we test against
- Inputs selected at launch time for:
    - driver-specific config (e.g. odsp endpoint)
    - console output
    - whether break points can be set in legacy modules (this is a fairly large performance hit, so default is 'no')
- Tinylicious launch target is a compound target which allows simultaneous server + client debugging
- FF code under `.legacy` (i.e. the modules installed for compat testing) isn't under skipFiles by default, so break points in this code works (either in the js if source map loading is turned off, or in the ts if it is turned on, which is configurable at launch time)

The r11s tests are difficult to run on Windows, so I didn't test that launch target. Other targets all work well as far as I can tell (which includes setting breakpoints in various places in dependent packages)

Sample view of launch window when using this workspace:

image

### CODE REVIEW

File: packages/test/test-end-to-end-tests/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"label": "Build Current Tests",
+			"type": "process",
+			"command": "node",
+			"args": [
+				"${workspaceRoot:FluidFramework}/node_modules/@fluidframework/build-tools/dist/fluidBuild/fluidBuild.js",
```
Comment (alexvy86) [inline packages/test/test-end-to-end-tests/.vscode/tasks.json:9]: Would this read cleaner with the first arg being `fluid-build` and adding the bit below? The current form assumes the root has its deps installed, and in that case calling `fluid-build` directly should work, no?

```
"options": {
  "cwd": "${workspaceRoot:FluidFramework}"
}
```

Comment (alexvy86) [reply packages/test/test-end-to-end-tests/.vscode/tasks.json:9]: And if that works, I guess remove the --root flag.

File: packages/test/test-end-to-end-tests/.vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,235 @@
+{
+	// Use IntelliSense to learn about possible attributes.
+	// Hover to view descriptions of existing attributes.
+	// For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+	"version": "0.2.0",
+	"configurations": [
```
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/.vscode/launch.json:6]: I've personally never really had an issue running e2e tests with different drivers or anything. I'm willing to try this out.

As for the performance hit for legacy tests, I have some draft that I think makes it way faster, but I haven't had time to get it production ready: https://github.com/microsoft/FluidFramework/pull/21881

File: packages/test/test-end-to-end-tests/.vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+				{ "label": "Yes", "value": "**/dummy-folder-unused/**" },
+			],
+		},
+		{
+			"id": "odspEndpoint",
+			"description": "Which odsp endpoint should be used?",
+			"type": "pickString",
+			"default": "odsp",
+			"options": ["odsp", "odsp-df"],
+		},
```
Comment (alexvy86) [inline packages/test/test-end-to-end-tests/.vscode/launch.json:233]: Just for reference, we could do something similar for r11s tests against local docker (the endpoint in that case is `docker` if I recall correctly) but it's such a rare case that I don't mind skipping it for now.

**Review by alexvy86 — APPROVED** [review]
Nice :). Didn't go line by line on all the launch targets but if anything is broken we can fix it on the go, this is a great improvement.

File: packages/test/test-end-to-end-tests/.vscode/tasks.json (hunk 1 above)
Comment (Abe27342) [reply packages/test/test-end-to-end-tests/.vscode/tasks.json:9]: can't be the first arg, it needs to be the command, but I believe we could do this if we wanted:

image

I copied this task from the root tasks.json (annoyingly we don't inherit the definitions there, just the launch targets AFAICT), so if we want to do this we should do it there as well.

**Review by tyler-cai-microsoft — APPROVED** [review]

File: packages/test/test-end-to-end-tests/.vscode/launch.json (hunk 2 above)
Comment (Abe27342) [reply packages/test/test-end-to-end-tests/.vscode/launch.json:6]: we definitely can do better perf-wise in the install step. The perf hit that this comment / option refers to is from a different place though: VSCode takes quite a while to load all of the `outFiles` in our monorepo if we're not careful. You can see this by using `"trace": true` in one of the launch targets and playing around with https://microsoft.github.io/vscode-pwa-analyzer/index.html.

Unsurprisingly, it's therefore a noticeable perf hit to tell vscode to load and make sure breakpoints get set in all of the extra versions of Fluid we install for our compat test matrix, especially when debugging e2e tests often doesn't require it.

File: packages/test/test-end-to-end-tests/.vscode/launch.json (hunk 3 above)
Comment (Abe27342) [reply packages/test/test-end-to-end-tests/.vscode/launch.json:234]: good idea. I can't test it, but added anyway in line with comments we have in root launch.json.

File: packages/test/test-end-to-end-tests/.vscode/tasks.json (hunk 1 above)
Comment (alexvy86) [reply packages/test/test-end-to-end-tests/.vscode/tasks.json:9]: Oh right, replace the node invocation entirely, yep.

**Pull request merged**